pkg/vcs/linux.go

```diff
@@ -214,37 +214,65 @@ func (ctx *linux) Bisect(bad, good string, trace io.Writer, pred func() (BisectR
 }
 
 func (ctx *linux) addMaintainers(com *Commit) {
-	if len(com.CC) > 2 {
+	if len(com.Recipients) > 2 {
 		return
 	}
-	list := ctx.getMaintainers(com.Hash, false)
-	if len(list) < 3 {
-		list = ctx.getMaintainers(com.Hash, true)
+	mtrs := ctx.getMaintainers(com.Hash, false)
+	if len(mtrs) < 3 {
+		mtrs = ctx.getMaintainers(com.Hash, true)
 	}
-	com.CC = email.MergeEmailLists(com.CC, list)
+	com.Recipients = append(com.Recipients, mtrs...)
+	sort.Sort(com.Recipients)
 }
 
-func (ctx *linux) getMaintainers(hash string, blame bool) []string {
+func (ctx *linux) getMaintainers(hash string, blame bool) Recipients {
 	// See #1441 re --git-min-percent.
 	args := "git show " + hash + " | " +
 		filepath.FromSlash("scripts/get_maintainer.pl") +
-		" --no-n --no-rolestats --git-min-percent=20"
+		" --git-min-percent=20"
 	if blame {
 		args += " --git-blame"
 	}
 	output, err := osutil.RunCmd(time.Minute, ctx.git.dir, "bash", "-c", args)
 	if err != nil {
 		return nil
 	}
-	var list []string
-	for _, line := range strings.Split(string(output), "\n") {
-		addr, err := mail.ParseAddress(line)
+	return ParseMaintainersLinux(output)
+}
+
+func ParseMaintainersLinux(text []byte) Recipients {
+	lines := strings.Split(string(text), "\n")
+	reRole := regexp.MustCompile(` \([^)]+\)$`)
+	var mtrs Recipients
+	// LMKL is To by default, but it changes to Cc if there's also a subsystem list.
+	lkmlType := To
+	lkmlCheck := false
+	for _, line := range lines {
+		role := reRole.FindString(line)
+		address := strings.Replace(line, role, "", 1)
+		addr, err := mail.ParseAddress(address)
 		if err != nil {
 			continue
 		}
-		list = append(list, strings.ToLower(addr.Address))
+		var roleType RecipientType
+		if strings.Compare(addr.Address, "linux-kernel@vger.kernel.org") == 0 {
```

Comment:
Isn't it possible to just write `if addr.Address == "linux-kernel@vger.kernel.org" {`?